internal/substatoptimizer/logger.go

```diff
@@ -0,0 +1,25 @@
+package substatoptimizer
```

Comment:
Given that most of the code have moved to `pkg/optimization` i think it's fine/better to move this package over as well. Or rather, if anything I think function *might* make more sense in the executable code since really this is just setting up and calling the optimization code